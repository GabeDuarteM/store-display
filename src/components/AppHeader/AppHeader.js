import React from 'react'
import PropTypes from 'prop-types'
import injectSheet from 'react-jss'

export const styles = {
  appHeader: {
    color: '#585858',
    fontFamily: 'Montserrat',
    fontSize: 24.0,
    fontWeight: 400,
    textAlign: 'center',
  },
}

const AppHeader = ({ classes }) => (
  <div className={classes.appHeader}>STORE DISPLAY</div>
)

export const propTypes = {
  /** @ignore */
  classes: PropTypes.object.isRequired,
}

AppHeader.propTypes = propTypes

export default injectSheet(styles)(AppHeader)
