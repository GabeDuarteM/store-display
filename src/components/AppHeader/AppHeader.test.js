import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import AppHeader, { styles, propTypes } from './AppHeader'

describe('COMPONENT: <AppHeader />', () => {
  it('should render without crashing', () => {
    shallow(<AppHeader />)
  })

  it("should match component's snapshot", () => {
    const wrapper = shallow(<AppHeader />)
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  it("should match component's style snapshot", () => {
    expect(styles).toMatchSnapshot()
  })

  it("should match component's propTypes snapshot", () => {
    expect(propTypes).toMatchSnapshot()
  })

  it('should have the text set to STORE DISPLAY', () => {
    const wrapper = shallow(<AppHeader />)
    expect(wrapper.dive().text()).toBe('STORE DISPLAY')
  })
})
