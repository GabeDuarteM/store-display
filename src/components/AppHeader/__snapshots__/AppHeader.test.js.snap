// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT: <AppHeader /> should match component's propTypes snapshot 1`] = `
Object {
  "classes": [Function],
}
`;

exports[`COMPONENT: <AppHeader /> should match component's snapshot 1`] = `
<div
  className="AppHeader-appHeader"
>
  STORE DISPLAY
</div>
`;

exports[`COMPONENT: <AppHeader /> should match component's style snapshot 1`] = `
Object {
  "appHeader": Object {
    "color": "#585858",
    "fontFamily": "Montserrat",
    "fontSize": 24,
    "fontWeight": 400,
    "textAlign": "center",
  },
}
`;
